File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int i = 0, j = s.size() - 1;
        int baseCase = s.size() % 2;
        while ((j - i) > baseCase) {
            char temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            ++i;
            --j;
        };
    };
};

File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        i = 0
        j = len(s)-1

        while (j-i>len(s)%2):
            temp = s[j]
            s[j] = s[i]
            s[i] = temp
            i += 1
            j -= 1